### PR TITLE
path.digits

### LINKED ISSUE #271: Configurable precision for d3.geoPath
As with d3.pathFixed https://github.com/d3/d3-path/pull/12, we should have a way to limit the precision of the strings generated by d3.geoPath. Currently that is implemented by the PathString class:

https://github.com/d3/d3-geo/blob/main/src/path/string.js

We should probably default to three decimal digits of precision, too.

### PR DESCRIPTION
**Pull request by mbostock**
Fixes #271. Related https://github.com/d3/d3-path/pull/12.

### CODE REVIEW

File: src/path/string.js [hunk 1]
```diff
@@ -1,59 +1,83 @@
...
+
+function appendFixed(digits) {
+  if (digits !== cacheDigits) {
+    (0).toFixed(digits); // validate digits
+    cacheDigits = digits;
+    cacheAppend = function append(strings) {
+      let i = 1;
+      this._ += strings[0];
+      for (const j = strings.length; i < j; ++i) {
+        this._ += +arguments[i].toFixed(digits) + strings[i];
```
Comment (mbostock) [inline src/path/string.js:78]: We might need additional coercion here if the stream is passed non-numeric input.

**Comment by mbostock** [comment]
I did some benchmarking in https://observablehq.com/d/07584fd387781785 and learned some things.

untitled (86)

* Switching to string concatenation is faster than the current _array_.join method. 👍 
* Using template literals, and even a custom template literal tag, is fast. 👍
* Coercing the result of toFixed to a number and then to a string to trim trailing zeroes is slow. 👎
* A custom function for trimming trailing zeroes is faster. 👍 
* But much faster still is to use Math.round instead. 👍👍

So, I’d like to use the Math.round approach. We learned six years ago in https://github.com/d3/d3-path/issues/10#issuecomment-262577521 that Math.round is faster than _number_.toFixed, so I’m just repeating that lesson here.

**Review by Fil — APPROVED** [review]
Nice!

**Comment by mbostock** [comment]
Related https://github.com/d3/d3-path/pull/11 which also adopted string concatenation over _array_.join for faster performance.

**Pull request merged**